Files/View Models/Properties/FileProperty.cs

```diff
@@ -0,0 +1,269 @@
+        /// </summary>
+        /// <param name="file"></param>
+        /// <returns></returns>
+        public async Task SaveValueToFile(StorageFile file)
+        {
+            if (!string.IsNullOrEmpty(Property)) 
+                return;
+
+            var propsToSave = new Dictionary<string, object>();
+            propsToSave.Add(Property, Converter.ConvertBack(Value, null, null, null));
+            await file.Properties.SavePropertiesAsync(propsToSave);
+        }
+
+        /// <summary>
+        /// Call this function just after getting properties to set the property's converter based on the value type.
+        /// For some reason, this does not work for arrays. In the case of arrays, override the converter
+        /// </summary>
+        private IValueConverter GetConverter()
+        {
+            if (Value is UInt32)
+                return new UInt32ToString();
+
+            if (Value is Double)
+                return new DoubleToString();
+
+            if (Value is DateTimeOffset)
+                return new DateTimeOffsetToString();
+
+            if(Value != null && Value.GetType().IsArray)
+            {
+                if (Value.GetType().GetElementType().Equals(typeof(string)))
+                    return new StringArrayToString();
+
+                if (Value.GetType().GetElementType().Equals(typeof(double)))
+                    return new DoubleArrayToString();
+            }
+            return null;
+        }
+
+        /// <summary>
+        /// Converts the property to a string to be displayed
+        /// </summary>
+        /// <returns></returns>
+        private string ConvertToString()
+        {
+            if (DisplayFunction != null)
+                return DisplayFunction.Invoke(Value);
+
+            if (Converter != null && Value != null)
+                return Converter.Convert(Value, typeof(string), null, null) as string;
+
+            return Value as string;
+        }
+
+        /// <summary>
+        /// Converts a string from a text block back to it's original type
+        /// </summary>
+        /// <param name="value">The input string</param>
+        /// <returns></returns>
+        private object ConvertBack(string value)
+        {
+            if (Converter != null && value != null)
+                return Converter.ConvertBack(value, typeof(object), null, null);
+
+            return value;
+        }
+
+        /// <summary>
+        /// Uses the name and section name resources to get their values
+        /// </summary>
+        public void InitializeNames()
+        {
+            Name = ResourceController.GetTranslation(NameResource);
+            Section = ResourceController.GetTranslation(SectionResource);
+        }
+
+        public static readonly List<FileProperty> PropertyListItemsBase = new List<FileProperty>()
+        {
+            new FileProperty() { NameResource = "PropertyAddress", SectionResource = "PropertySectionGPS", ID = "address" },
```

Comment:
Assign the properties in a constructor?

Reply:
Actually instead of hardcoding it like this, I'd create a factory class that'd instantiate FileProperty from databse